### ISSUE TITLE
No option to enable Built-in CODE Server

### ISSUE THREAD

**Issue by idreamerhx** [CollaboraOnline/richdocumentscode#54]
I setup up Nextcloud 19.0.1 on a rk3399 (aarch64, archlinuxarm) board. the nextcould works fine. Because of low network speed, i download richdocumentscode package manually, unzip into nextcloud/apps and chown to http. Enabled in nextcould panel.

in setting  ->  Collabora Online
There is only two options:
Use your own server
Use a demo server

So, where is or how to enable "Built-in CODE Server"

Thanks.

**Comment by keunes** [comment]
I the issue that the built-in CODE server is not appearing in the Collabora Online settings in Nextcloud. The difference with the OP is that I'm running Raspberry Pi OS Lite a Raspberry Pi 4, and that I installed both the CODE server and Collabora Online through the Nextcloud web-interface app store.

[Here](https://www.reddit.com/r/raspberry_pi/comments/7hckbr/using_appimage_on_raspberry_pi/dqrorhn/?utm_source=reddit&utm_medium=web2x&context=3) I read that the AppImage needs to have support for ARM-based platforms, in order to run. Is this true? And, if yes, does the CODE Server AppImage support ARM-based systems?

When trying to run the CODE Server from the command line with `sudo -u www-data Collabora_Online.AppImage`
I get the following error: `sudo: Collabora_Online.AppImage: command not found`

**Comment by diode-ee** [comment]
I'm in the same boat. After installing and going to settings, I do not see the option to run the internal server. I'm running 19.0.2 on a TrueNAS jail. I've tried removing and reinstalling with no joy.

**Comment by keunes** [comment]
@diode-ee I think my issue is trying to run it on a Raspberry Pi, which is an ARM device (not supported). Does your TrueNAS jail have an intel chip?

**Comment by hyphone** [comment]
@keunes ARM is not supported even if this [package](https://apps.nextcloud.com/apps/richdocumentscode_arm64) is available?

**Comment by keunes** [comment]
@hyphone Oh? What do you mean?

**Comment by bcutter** [comment]
I installed the apps "Collabora Online" and when trying to install "Collabora Online - Built-in CODE Server (ARM64)" on the second try it gives error

```
 cURL error 28: Operation timed out after 120000 milliseconds with 373971277 out of 410601603 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) 
```

The first time installation seems to finish and it looked like here https://github.com/CollaboraOnline/richdocumentscode/issues/54#issuecomment-687702556. Obviously the app wasn´t installed properly which is the reason why it´s not showing up as an option on the https://my-nextcloud-server.org/index.php/settings/admin/richdocuments page.

**Comment by hyphone** [comment]
> @hyphone What do you mean? And was that a rhetorical or actual question?

It was an actual question as you say ARM is not supported but there is an ARM64 package for it. And I'm a bit confused - when it's not supported - what is the package for?

>  cURL error 28: Operation timed out after 120000 milliseconds with 373971277 out of 410601603 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) 

I installed it via command line and didn't get the timeout. I can select the  Built-in CODE Server but Collabora cannot connect to it.

**Comment by keunes** [comment]
> It was an actual question as you say ARM is not supported but there is an ARM64 package for it.

I think the ARM package was provided between when I made my comment and when you replied.

> And I'm a bit confused - when it's not supported - what is the package for?

Please see here: https://github.com/CollaboraOnline/richdocumentscode/issues/58#issuecomment-704105620
TLDR; It's in the works but not finished yet thus not officially supported yet.
